### PR TITLE
CMP-4564: exposed new openCurrentUserStatusTransaction function

### PR DESCRIPTION
**Pull request by felipe-saez**
Exposed new `openCurrentUserStatusTransaction` function.
This function contains method used to enable and disable vendors and purposes.

### CODE REVIEW

File: android/src/main/kotlin/io/didomi/fluttersdk/DidomiPlugin.kt [hunk 1]
```diff
@@ -664,4 +666,40 @@ class DidomiPlugin : FlutterPlugin, MethodCallHandler, ActivityAware {
...
+     * Here all the changes to the user status related to purposes and vendors are applied and committed at once.
+     */
+    private fun commitCurrentUserStatusTransaction(call: MethodCall, result: Result) {
+        try {
+            val enabledPurposes = argumentAsArray("enabledPurposes", call)
+            val disabledPurposes = argumentAsArray("disabledPurposes", call)
+            val enabledVendors = argumentAsArray("enabledVendors", call)
+            val disabledVendors = argumentAsArray("disabledVendors", call)
+
+            val transaction = Didomi.getInstance().openCurrentUserStatusTransaction()
```
Comment (pmerlet-at-didomi) [inline android/src/main/kotlin/io/didomi/fluttersdk/DidomiPlugin.kt:680]:
```
val updated = Didomi.getInstance().openCurrentUserStatusTransaction
    .enablePurposes(*enabledPurposes)
    .disablePurposes(*disabledPurposes)
    .enableVendors(*enabledVendors)
    .disableVendors(*disabledVendors)
    .commit()
```

File: ios/Classes/SwiftDidomiSdkPlugin.swift [hunk 2]
```diff
@@ -887,4 +889,24 @@ public class SwiftDidomiSdkPlugin: NSObject, FlutterPlugin {
...
+        guard let args = call.arguments as? Dictionary<String, [String]> else {
+            result(FlutterError.init(code: "invalid_args", message: "Wrong arguments for commitCurrentUserStatusTransaction", details: nil))
+            return
+        }
+
+        if !Didomi.shared.isReady() {
+            result(FlutterError.init(code: "sdk_not_ready", message: SwiftDidomiSdkPlugin.didomiNotReadyException, details: nil))
+            return
+        }
+        let transaction = Didomi.shared.openCurrentUserStatusTransaction()
```
Comment (pmerlet-at-didomi) [inline ios/Classes/SwiftDidomiSdkPlugin.swift:905]:
```
let updated = Didomi.shared.openCurrentUserStatusTransaction()
       .enablePurposes(args["enabledPurposes"] ?? [])
       .disablePurposes(args["disabledPurposes"] ?? [])
       .enableVendors(args["enabledVendors"] ?? [])
       .disableVendors(args["disabledVendors"] ?? [])
        .commit()
result(updated)
```

File: lib/transactions/current_user_status_transaction.dart [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+/// Class used to contain all the operations related to transactions.
+/// All the changes regarding the status of purposes and vendors will
+/// only be communicated to the native SDKs once the `commit` method is called.
+class CurrentUserStatusTransaction {
+  final Map<String, bool> purposesStatus = {};
+  final Map<String, bool> vendorsStatus = {};
+
+  /// Callback used to communicate the updates made through this transaction to
+  /// the native SDKs.
+  final NativeCallback nativeCallback;
```
Comment (pmerlet-at-didomi) [inline lib/transactions/current_user_status_transaction.dart:13]: Is there a way to hide this from client code, like a `private` field?

File: lib/transactions/current_user_status_transaction.dart [hunk 4]
```diff
@@ -0,0 +1,74 @@
+typedef NativeCallback = Future<bool> Function(
+    List<String> enabledPurposes, List<String> disabledPurposes, List<String> enabledVendors, List<String> disabledVendors);
+
+/// Class used to contain all the operations related to transactions.
+/// All the changes regarding the status of purposes and vendors will
+/// only be communicated to the native SDKs once the `commit` method is called.
+class CurrentUserStatusTransaction {
+  final Map<String, bool> purposesStatus = {};
```
Comment (pmerlet-at-didomi) [inline lib/transactions/current_user_status_transaction.dart:8]: If possible, we should prevent clients from modifying this directly

**Review by pmerlet-at-didomi — APPROVED** [review]

File: android/src/main/kotlin/io/didomi/fluttersdk/DidomiPlugin.kt (hunk 1 above)
Comment (felipe-saez) [reply android/src/main/kotlin/io/didomi/fluttersdk/DidomiPlugin.kt]: done

File: ios/Classes/SwiftDidomiSdkPlugin.swift (hunk 2 above)
Comment (felipe-saez) [reply ios/Classes/SwiftDidomiSdkPlugin.swift]: done

File: lib/transactions/current_user_status_transaction.dart (hunk 4 above)
Comment (felipe-saez) [reply lib/transactions/current_user_status_transaction.dart]: I've added `_` as a prefix which makes fields private.

**Review by pmerlet-at-didomi — DISMISSED** [review]

File: example/lib/widgets/transactions/enable_purposes_transaction.dart [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  Future<String> callDidomiPlugin() async {
+    final transaction = await DidomiSdk.openCurrentUserStatusTransaction();
+    final entity = "cookies";
+    transaction.enablePurposes([entity]);
+    final updated = await transaction.commit();
+    final status = await DidomiSdk.currentUserStatus;
+    final enabled = status.purposes?[entity]?.enabled;
+    return "Updated: $updated, Enabled: $enabled.";
+  }
+}
```
[the remark this answers is not in the document]
Comment (felipe-saez) [reply example/lib/widgets/transactions/enable_purposes_transaction.dart]: I believe I've fixed all the "new line" issues.

**Review by pmerlet-at-didomi — APPROVED** [review]

**Review by pmerlet-at-didomi — APPROVED** [review]

**Review by pmerlet-at-didomi — APPROVED** [review]

**Pull request merged**